rc/windowing/detection.kak

```diff
@@ -0,0 +1,63 @@
+# Attempt to detect the windowing environment we're operating in
+#
+# We try to load modules from the windowing_modules str-list option in order,
+# stopping when one of the modules loads successfully. This ensures that only
+# a single module is loaded by default.
+#
+# On load each module must attempt to detect the environment it's appropriate
+# for, and if the environment isn't appropriate it must fail with an error.
+# In addition, each module must check for the length of the windowing_modules
+# str-list option defined below, and must /not/ check for an appropriate
+# environment if the list is empty. An example of this test:
+#
+# evaluate-commands %sh{
+#     set -- ${kak_opt_windowing_modules}
+#     [ $# -eq 0 ] || [ -n "$TMUX" ] || echo 'fail tmux not detected'
```

Comment:
`[ -z "$kak_opt_windowing_modules" ]` would work as well, avoiding the need to the `set` line.

More generally, I wonder about using that option in each module. It seems like convoluted way to force loading of modules, I guess it can be good enough for now, but not entirely satisfying.

Reply:
The two ideas "enable/disable environment detection" and "load modules in this order" are pretty orthogonal, and we could configure them separately with different options, but then we have to decide how those two options would interact, and we're right back into the weeds we were in at the start of the discussion. That's why I've been trying to focus on the user experience rather than the abstract logic - almost all users will want a single module loaded which is relevant to their current environment, and anything else is just going to be confusing. Anyone who /doesn't/ want that is likely to be doing something so weird that there's no point us trying to implement anything to support it, and the best option is going to be getting completely out of the way and letting them do whatever they want. Using an empty list to indicate that is overloading the meaning a /little/, but only a little, and it makes for a simple implementation as well as an interface that's pretty easy to understand even if it's not perfectly elegant.